commit efba82fc7c6f4c4ef36e284b45c6e222e72b4262
Author:     Isaac Yang <isaacy@nvidia.com>
AuthorDate: Wed Sep 13 07:11:31 2023 -0700
Commit:     Isaac Yang <isaacy@nvidia.com>
CommitDate: Wed Sep 13 13:47:36 2023 -0700

    Add secure argument to Task and set secure argument in cell's call.
---
 nvflare/private/fed/server/server_engine.py | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index cb56834d0..1d238532a 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -497,7 +497,14 @@ class ServerEngine(ServerEngineInternalSpec):
         self.run_manager.aux_runner.register_aux_message_handler(topic, message_handle_func)
 
     def send_aux_request(
-        self, targets: [], topic: str, request: Shareable, timeout: float, fl_ctx: FLContext, optional=False
+        self,
+        targets: [],
+        topic: str,
+        request: Shareable,
+        timeout: float,
+        fl_ctx: FLContext,
+        optional=False,
+        secure=False,
     ) -> dict:
         try:
             if not targets:
@@ -506,7 +513,13 @@ class ServerEngine(ServerEngineInternalSpec):
                     targets.append(t.name)
             if targets:
                 return self.run_manager.aux_runner.send_aux_request(
-                    targets=targets, topic=topic, request=request, timeout=timeout, fl_ctx=fl_ctx, optional=optional
+                    targets=targets,
+                    topic=topic,
+                    request=request,
+                    timeout=timeout,
+                    fl_ctx=fl_ctx,
+                    optional=optional,
+                    secure=secure,
                 )
             else:
                 return {}
